src/Search.jsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useMemo, useRef } from "react"
+import { searchStore } from "./App";
+import './Search.css'
+
+export function Search () {
```

Comment:
you can also make this component generic and you can pass the input setter and input value via props, try to do that